datastore/postgres.go

```diff
@@ -11,24 +12,29 @@ import (
 const tableName = "records"
 
 // NewPostgreSQLDatastore creates a new pgds.Datastore that talks to a PostgreSQL database
-func NewPostgreSQLDatastore(ctx context.Context, connstr string) (*pgds.Datastore, error) {
-	connConf, err := pgxpool.ParseConfig(connstr)
-	if err != nil {
-		return nil, err
-	}
-	pool, err := pgxpool.ConnectConfig(ctx, connConf)
-	if err != nil {
-		return nil, err
-	}
-	_, err = pool.Exec(ctx, fmt.Sprintf("CREATE TABLE IF NOT EXISTS %s (key TEXT NOT NULL UNIQUE, data BYTEA)", tableName))
-	if err != nil {
-		return nil, err
-	}
-	_, err = pool.Exec(ctx, fmt.Sprintf("CREATE INDEX IF NOT EXISTS %s_key_text_pattern_ops_idx ON %s (key text_pattern_ops)", tableName, tableName))
-	if err != nil {
-		return nil, err
+func NewPostgreSQLDatastore(ctx context.Context, connstr string, createDB bool) (*pgds.Datastore, error) {
+	if createDB {
+		connConf, err := pgxpool.ParseConfig(connstr)
+		if err != nil {
+			return nil, err
+		}
+		pool, err := pgxpool.ConnectConfig(ctx, connConf)
+		if err != nil {
+			return nil, err
+		}
+		fmt.Fprintf(os.Stderr, "Creating Table\n")
+		_, err = pool.Exec(ctx, fmt.Sprintf("CREATE TABLE IF NOT EXISTS %s (key TEXT NOT NULL UNIQUE, data BYTEA)", tableName))
+		if err != nil {
+			return nil, err
+		}
+		fmt.Fprintf(os.Stderr, "Creating Index\n")
+		_, err = pool.Exec(ctx, fmt.Sprintf("CREATE INDEX IF NOT EXISTS %s_key_text_pattern_ops_idx ON %s (key text_pattern_ops)", tableName, tableName))
+		if err != nil {
+			return nil, err
+		}
+		pool.Close()
 	}
-	pool.Close()
+	fmt.Fprintf(os.Stderr, "Connecting to Database\n")
```

Comment:
Shall we remove these `fmt.Fprintf` logs or do you think they're useful?

Reply:
It was hanging at the 2nd of the three things we do here, nice to know where it is I think